libraries/AP_TECS/AP_TECS.cpp

```diff
@@ -916,6 +916,12 @@ void AP_TECS::_update_throttle_without_airspeed(int16_t throttle_nudge, float pi
 
 void AP_TECS::_detect_bad_descent(void)
 {
+    // Don't detect bad descents when gliding or transitioning
```

Comment:
I would move the _flags.underspeed check to this line, eg

```
// Don't detect bad descents when gliding or transitioning or when recovering from underspeed
if (_flags.is_gliding || _flight_stage == AP_FixedWing::FlightStage::VTOL || _flags.underspeed) {
```

Reply:
Done